### PR TITLE
SO-4924: RF2 import improvements

### PR DESCRIPTION
**Pull request by cmark**
This PR add the following functionality to the RF2 import API:
* Support smart autogeneration of delta content when importing Full releases (the existing component-based effective time based filtering is still there, but this makes the import much faster and easier to understand)
* Support `importUntil` date property to import only a portion from an RF2 release
* Disallow importing RF2 releases with `createVersions` enabled when the target branch is a CodeSystem's main development branch and it has unpublished changes.
* Update test RF2 package to 2021-07-31
* Minor fixes

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2NextReleaseImportTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+public class SnomedRf2NextReleaseImportTest extends AbstractSnomedApiTest {
+
+	@Test
+	public void import01_CurrentReleaseAgain_NoChanges() throws Exception {
+		doImport(Resources.Snomed.MINI_RF2_INT_20210731, "20210131");
+	}
+
+	@Test
+	public void import02_NextFullRelease_20210731() throws Exception {
+		doImport(Resources.Snomed.MINI_RF2_INT_20210731, "20210731");
```
Comment (nagyo) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2NextReleaseImportTest.java:55]: Would be nice to have some sort of assertion that the 20210731 content is indeed there. At this point we only know that the import job was successful, but nothing about the content itself

Comment (nagyo) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2NextReleaseImportTest.java:55]: or maybe check that the version branch exists?

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2ImportRestService.java [hunk 2]
```diff
@@ -87,10 +87,14 @@
...
 			final Boolean createVersions,
 			
-			@Parameter(description = "Import should be allowed to progress when members of listed reference sets have missing referenced components")
+			@Parameter(description = "Ignore missing referenced components in listed references instead of reporting them as an error.")
 			@RequestParam(name = "ignoreMissingReferencesIn", required = false)
 			final List<String> ignoreMissingReferencesIn,
 			
+			@Parameter(description = "Import all component until this specified effective time value", schema = @Schema(format = "date", pattern = "\\d{6}"))
+			@RequestParam(name = "importUntil", required = false)
+			final String importUntil,
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2ImportRestService.java:96]: This parameter is not propagated to the import request below.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/MapDBRf2EffectiveTimeSlice.java [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+				Set<String> requiredDependencies = LongSets.toStringSet(dependencies);
+				for (String requiredDependency : requiredDependencies) {
+					dependenciesByComponent.put(getCdoType(requiredDependency), requiredDependency);
+				}
+			}
+		}
+		return dependenciesByComponent;
+	}
+
+	private Class<? extends SnomedDocument> getCdoType(String componentId) {
```
Comment (nagyo) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/MapDBRf2EffectiveTimeSlice.java]: Would be nice to rename this class 🤢

**Review by nagyo — CHANGES_REQUESTED** [review]
LGTM beside the few comments I made

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2NextReleaseImportTest.java (hunk 1 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2NextReleaseImportTest.java:55]: Yes, we could verify the ROOT concept's description or as a minimum the version itself.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/MapDBRf2EffectiveTimeSlice.java (hunk 3 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/MapDBRf2EffectiveTimeSlice.java]: Renamed :)

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2NextReleaseImportTest.java (hunk 1 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2NextReleaseImportTest.java:55]: Fixed by a version check.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2EffectiveTimeSlices.java [hunk 4]
```diff
@@ -45,20 +46,33 @@
...
+		this.latestVersionEffectiveTime = latestVersionEffectiveTime;
+		this.importUntil = importUntil;
 	}
 	
 	public Rf2EffectiveTimeSlice getOrCreate(String effectiveTime) {
 		if (!slices.containsKey(effectiveTime)) {
-			slices.put(effectiveTime, new Rf2EffectiveTimeSlice(db, effectiveTime, loadOnDemand));
+			// if the incoming effectiveTime value is greater than or equal to the current release, then allow reading
+			if (effectiveTime.compareTo(latestVersionEffectiveTime) > 0) {
+				if ((Strings.isNullOrEmpty(importUntil) || effectiveTime.compareTo(importUntil) <= 0)) {
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2EffectiveTimeSlices.java:63]: The `LocalDate` in the request is converted using `EffectiveTimes` at some point. Are you sure `importUntil` is not set to `"Unpublished"` in such cases?

**Review by apeteri — APPROVED** [review]

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2EffectiveTimeSlices.java (hunk 4 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2EffectiveTimeSlices.java:63]: Yes, that can only happen when `importUntil` is null, but in that case we propagate `null` value instead.
https://github.com/b2ihealthcare/snow-owl/pull/918/files#diff-552abddd6cab4bf2cd5bfb7bd2207476152dcd20183245fc7d11fa3e3399462dR74

Comment (apeteri) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2EffectiveTimeSlices.java:63]: (Although `"9999-12-31`" should be smaller or equal than `"Unpublished"` 😄)

**Review by nagyo — APPROVED** [review]
LGTM!

**Pull request merged**